tests/pkcs7_tests.cpp

```diff
@@ -0,0 +1,177 @@
+        "l+Ht0Bi8/K5jWDolMogg2BALlWuKPrJY5GTn8jyFE1V1LE063E7x2qa+Wu4MSV7S\n"
+        "8JZfM+LdWy7/ygxpzcBqpxxKaDo0A/XPW4W6pTHPPt2U4sLstvQvlfAP09AM2n5P\n"
+        "8em9JI2ugTzTfv2eKh9YdYfDjFAs5P9+7u4SZ+z94jS4ydixtkyRxFzhGC6PSaQN\n"
+        "m2pO38lpsE9jAsc2DKmg+LO+GwXq2RkmF7fAXGHe2cbbYAh5TwGzqwHSnBfNV7vU\n"
+        "rF/IzSWCpw3g07+UMWwg4TCCBd0wggPFoAMCAQICEQC8/E7Ywbq5XuYt5s12X5NC\n"
+        "MA0GCSqGSIb3DQEBCwUAMHkxCzAJBgNVBAYTAlVTMRIwEAYDVQQHEwlDaGFybG90\n"
+        "dGUxEzARBgNVBAoTCk5ldGZvdW5kcnkxEDAOBgNVBAsTB0FEVi1ERVYxLzAtBgNV\n"
+        "BAMTJk5ldGZvdW5kcnksIEluYy4gQ2VydGlmaWNhdGUgQXV0aG9yaXR5MB4XDTE5\n"
+        "MDkwMzE4MzQ0N1oXDTIwMDkwMjE4MzU0NFowczELMAkGA1UEBhMCVVMxEjAQBgNV\n"
+        "BAcTCUNoYXJsb3R0ZTETMBEGA1UEChMKTmV0Zm91bmRyeTEQMA4GA1UECxMHQURW\n"
+        "LURFVjEpMCcGA1UEAxMgTmV0Zm91bmRyeSwgSW5jLiBJbnRlcm1lZGlhdGUgQ0Ew\n"
+        "ggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDs36VUJANvDtYn8RA51i8A\n"
+        "97G06Id0yuOv5sDooT5a9qJRusCZcH8+ZBGcDzim5jkmruEOPsvU9ZdfoOgqtkgy\n"
+        "HXlziwPvjFf1QVaNDCQ3FiKOsDIzhMOIo8wjaGKikCF2Gj5bLlnAcxzLNJ9rqAnI\n"
+        "m5ggT376I1vadOl1UMUsThqg/rlnEUzZnkw5IbutrEUlPLKjmny5O+5dwrnR12Xh\n"
+        "KhNx78b2jbBIJ3+6hzYg3qH4RJ4RsPs/M44IPBvXErYXc0rPiFUXzkHmQtfFNE5v\n"
+        "4Awtix0u0F6HC+QXX7zJiO+Pyo0c35ttkW4+TlS9hfHrK2ooFYY0tCZfZ8rFKgVZ\n"
+        "r1hZPhrFDnSztxIzNTXHKgSj5vfUMzGhUuxWfR6jZit0y8wUWXYI7Ae8ECMAy0zW\n"
+        "JtKmzMXPOfWP+JXSckE2q1OxE5okc/dwoc6FsbqP/jHtvrugy/5wKPrUzCVHUNF1\n"
+        "sOGB6cnmfSNmlw4gZChrOacXFmup0qN8CV6y4kIteMqJrJKzIw8YssILe1H1eWH3\n"
+        "yVnPJfakRduCZWDxPaO5ml5oFqMx4pFxwGiBXhopZY+5HROhOTXFuptW+tzTYIfb\n"
+        "J7ARZ5omymuyTxmipoVzyKtjMqZY1Ftldcq60nG1r+8IRnewHMe2jabViiqB0kHK\n"
+        "krz8+0zUGs8iO1ysGo+DgQIDAQABo2YwZDAOBgNVHQ8BAf8EBAMCAYYwEgYDVR0T\n"
+        "AQH/BAgwBgEB/wIBADAdBgNVHQ4EFgQUiZi3d5QP62EW+moHPoEtgD1/utMwHwYD\n"
+        "VR0jBBgwFoAUBbwjE6D7YyjIh2njeqakqT2AClcwDQYJKoZIhvcNAQELBQADggIB\n"
+        "AHb01ZIjvW2scssqqudxNpz2ZgOvrCK26gBLq8E0e/eccpB1DtsNoAZNzE1aMqni\n"
+        "KvhG1Z8b6Jj+AEdbOrf+pjT62sB9QkrLvlhBlfeSeGKoYkxe9AgrTWGKNQ8QrV+I\n"
+        "ixpAgb6i+Zr/f7C9W9bgSKv3mCfTk2xS3DiC99j3JddDvNaDazd0eboKdFXUADoM\n"
+        "kis18E1/nyN+15zmeSQ09vv46nx9X1QZqxkFNHkfQf/c9q1ztp31zyO5PgSROfh4\n"
+        "f7ab4EKe2m9Ff4PLwavwJH75Ao41uTDtFncf0Hrl82Al6v2sI96u9Xi2ZN8xmLtO\n"
+        "UJt7VJezoBEaeVzbBAI85o+XEnOXXo5mI+X/IhujpEuuAIwdwiU6bGubEwHUySpf\n"
+        "CqezFwuLzZf6TE7SEa51i5W70KcXYmK1h+E9VqspHVBLkP6NbUSpQz50bYq58lT7\n"
+        "y7QKoJPEfsdtKDvquORL6r1QWvorV6mTMulnzVseOX0BShoAJylXjGww1oBhfhby\n"
+        "pSyX32tccwqCKIFz/a8GYcvdrjJquBmLVJ2a4hQl8p1RLnFY6T5nymlpGTXojTgk\n"
+        "/CovbcktdVivi8k+RC/KZZbq6IDTONRGsUrzOqUKsi7PkN685ML0pAaPEgHdr23y\n"
+        "fcwJ0v2IisYTCMavk0DJSj9Hd+coMSyTa7ghp8ja/0PSoQAxAA==";
+
+#define OID_PKCS7 MBEDTLS_OID_PKCS "\x07"
+#define OID_PKCS7_DATA OID_PKCS7 "\x02"
+#define OID_PKCS7_SIGNED_DATA OID_PKCS7 "\x01"
+
+TEST_CASE("parse pkcs7") {
```

Comment:
most of this code should be extracted and put into library and used:
something like: `pkcs7_parse_certs(char* pkcs7, size_t pkcs7len, mbedtls_x509_crt *chain);`

Reply:
you'd want to use your `extract_well_known_certs()` here, rigth?

Reply:
Done.

Reply:
sweet